uwblueprint/planet-read
# Soft delete users

## Notion ticket link



[Implement soft delete user mutation ](https://www.notion.so/uwblueprintexecs/Implement-soft-delete-user-mutation-6ec505fc7691411b9cea0a9c1d1afc5a)



## Implementation description

- Added database changes and migrations to create a users_all table and a users view table 
- Added soft delete mutation that also sets any story translation with the user translator/reviewer id to `null`, and soft deletes the comment that 



## Steps to test

**Test migration**
1. Rebuild docker containers and exec into backend container: `docker exec -it planet-read_py-backend_1 /bin/bash`
2. Clear db: `flask db downgrade base` / `python -m tools.insert_test_data erase`
3. Upgrade db: `flask db upgrade`
4. Seed db: `python -m tools.insert_test_data`

**Test delete**
1. Get access token by logging in as an admin and update headers in Altair
2. Soft delete a story translation:
3. 
```
mutation {
  softDeleteUser(id:1) {
    ok
  }
}
```

- check that the user doesn't show up in query `select * from users where id = 1;`, and that it does show up with is_deleted=1  (`select is_deleted from users_all where id =1;`)
- check that story_translations where the user was assigned is now `NULL` (from seed this should be rows 10 and 13)
- check that the comments that belonged to the user are soft deleted (`select is_deleted from comments_all where user_id = 1;`);



## What should reviewers focus on?

- Does the migration work?
- Does the soft delete for users work as expected, and update the correct values?

## Checklist
